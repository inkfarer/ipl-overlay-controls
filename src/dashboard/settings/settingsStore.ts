import { NodeCGBrowser } from 'nodecg/browser';
import { LastFmSettings, RadiaSettings, RuntimeConfig } from 'schemas';
import { GameVersion } from 'types/enums/gameVersion';
import { SetGameVersionResponse } from 'types/messages/runtimeConfig';
import { defineStore } from 'pinia';

const lastFmSettings = nodecg.Replicant<LastFmSettings>('lastFmSettings');
const radiaSettings = nodecg.Replicant<RadiaSettings>('radiaSettings');
const runtimeConfig = nodecg.Replicant<RuntimeConfig>('runtimeConfig');

export const settingsReps = [lastFmSettings, radiaSettings, runtimeConfig];

export interface SettingsStore {
    lastFmSettings: LastFmSettings
    radiaSettings: RadiaSettings
    runtimeConfig: RuntimeConfig
}

export const useSettingsStore = defineStore('settings', {
    state: () => ({
        lastFmSettings: {},
        radiaSettings: {
            guildID: null,
            enabled: null,
            updateOnImport: null
        },
        runtimeConfig: null
    } as SettingsStore),
    actions: {
        setLastFmSettings({ newValue }: { newValue: LastFmSettings }): void {
            lastFmSettings.value = newValue;
        },
        setRadiaSettings({ newValue }: { newValue: RadiaSettings }): void {
            radiaSettings.value = newValue;
        },
        setUpdateOnImport(updateOnImport: boolean): void {
            radiaSettings.value.updateOnImport = updateOnImport;
        },
        async attemptRadiaConnection(): Promise<void> {
            return nodecg.sendMessage('retryRadiaAvailabilityCheck');
        },
        setGameVersion(newValue: GameVersion): Promise<SetGameVersionResponse> {
            return nodecg.sendMessage('setGameVersion', { version: newValue });
        }
    }
});
