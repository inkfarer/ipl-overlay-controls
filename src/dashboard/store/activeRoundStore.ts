import { ActiveRound, SwapColorsInternally } from 'schemas';
import { createStore, Store, useStore } from 'vuex';
import cloneDeep from 'lodash/cloneDeep';
import { InjectionKey } from 'vue';
import { GameWinner } from 'types/enums/gameWinner';
import { SetActiveColorRequest, SetWinnerRequest, SwapRoundColorRequest } from 'types/messages/activeRound';
import { SetRoundRequest } from 'types/messages/rounds';
import { ActiveRoundGame } from 'types/activeRoundGame';

const activeRound = nodecg.Replicant<ActiveRound>('activeRound');
const swapColorsInternally = nodecg.Replicant<SwapColorsInternally>('swapColorsInternally');

export const activeRoundReps = [ activeRound, swapColorsInternally ];

export interface ActiveRoundStore {
    activeRound: ActiveRound;
    swapColorsInternally: SwapColorsInternally;
}

export const activeRoundStore = createStore<ActiveRoundStore>({
    state: {
        activeRound: null,
        swapColorsInternally: null
    },
    mutations: {
        setState(store, { name, val }: { name: string, val: unknown }): void {
            this.state[name] = cloneDeep(val);
        }
    },
    actions: {
        setWinner(store, { winner }: { winner: GameWinner }): void {
            nodecg.sendMessage('setWinner', { winner });
        },
        removeWinner(): void {
            nodecg.sendMessage('removeWinner');
        },
        setActiveColor(store, request: SetActiveColorRequest): void {
            nodecg.sendMessage('setActiveColor', request);
        },
        swapColors(): void {
            swapColorsInternally.value = !swapColorsInternally.value;
        },
        setWinnerForIndex(store, { index, winner }: { index: number, winner: GameWinner }): void {
            nodecg.sendMessage('setWinner', { winner, roundIndex: index } as SetWinnerRequest);
        },
        resetActiveRound(): void {
            nodecg.sendMessage('resetActiveRound');
        },
        setActiveRound(store, request: SetRoundRequest): void {
            nodecg.sendMessage('setActiveRound', request);
        },
        swapRoundColor(store, request: SwapRoundColorRequest): void {
            nodecg.sendMessage('swapRoundColor', request);
        },
        updateActiveGames(store, games: ActiveRoundGame[]): void {
            nodecg.sendMessage('updateActiveGames', { games });
        },
        switchToNextColor(): void {
            nodecg.sendMessage('switchToNextColor');
        },
        switchToPreviousColor(): void {
            nodecg.sendMessage('switchToPreviousColor');
        },
        async getNextAndPreviousColors(): Promise<void> {
            return nodecg.sendMessage('getNextAndPreviousColors');
        }
    }
});

export const activeRoundStoreKey: InjectionKey<Store<ActiveRoundStore>> = Symbol();

export function useActiveRoundStore(): Store<ActiveRoundStore> {
    return useStore(activeRoundStoreKey);
}
