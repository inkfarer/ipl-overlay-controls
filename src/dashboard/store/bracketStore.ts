import { Bracket } from '@tourneyview/common';
import { defineStore } from 'pinia';

const bracketData = nodecg.Replicant<Bracket | undefined>('bracketData');

export const bracketReps = [ bracketData ];

export interface BracketStore {
    bracketData: Bracket | undefined;
}

export const useBracketStore = defineStore('brackets', {
    state: () => ({
        bracketData: undefined
    } as BracketStore),
    actions: {
        resetBracketData() {
            bracketData.value = undefined;
        }
    }
});
