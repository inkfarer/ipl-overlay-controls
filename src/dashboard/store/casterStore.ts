import { NodeCGBrowser } from 'nodecg/browser';
import { Caster, Casters, RadiaSettings } from 'schemas';
import { generateId } from '../../helpers/generateId';
import { defineStore } from 'pinia';

const casters = nodecg.Replicant<Casters>('casters');
const radiaSettings = nodecg.Replicant<RadiaSettings>('radiaSettings');

export const castersReps = [ casters, radiaSettings ];

export interface CasterStore {
    casters: Casters;
    uncommittedCasters: Casters;
    radiaSettings: RadiaSettings;
}

const defaultCaster = {
    name: 'New Caster',
    twitter: '',
    pronouns: '?/?'
};

export const useCasterStore = defineStore('casters', {
    state: () => ({
        casters: {},
        uncommittedCasters: {},
        radiaSettings: {
            enabled: null,
            guildID: null,
            updateOnImport: null
        }
    } as CasterStore),
    actions: {
        updateCaster({ id, newValue }: { id: string, newValue: Caster }): void {
            casters.value[id] = newValue;
        },
        removeUncommittedCaster(id: string): void {
            delete this.uncommittedCasters[id];
        },
        addUncommittedCaster({ id, caster }: { id: string, caster?: Caster }): void {
            this.uncommittedCasters[id] = caster ?? defaultCaster;
        },
        addDefaultCaster(): string {
            const id = generateId();
            this.addUncommittedCaster({ id });
            return id;
        },
        async saveUncommittedCaster({ id, caster }: { id: string, caster: Caster }): Promise<string> {
            const newId = await nodecg.sendMessage('saveCaster', caster);
            this.removeUncommittedCaster(id);
            return newId;
        },
        async loadCastersFromVc(): Promise<void> {
            const result = await nodecg.sendMessage('getLiveCommentators');
            if (result.extra && result.extra.length > 0) {
                for (let i = 0; i < result.extra.length; i++) {
                    const extraCaster = result.extra[i];
                    const id = extraCaster.discord_user_id;
                    delete extraCaster.discord_user_id;
                    this.addUncommittedCaster({ id, caster: extraCaster });
                }
            }
        },
        showCasters() {
            nodecg.sendMessage('mainShowCasters');
        },
        async removeCaster(id: string): Promise<void> {
            return nodecg.sendMessage('removeCaster', { id });
        }
    }
});
