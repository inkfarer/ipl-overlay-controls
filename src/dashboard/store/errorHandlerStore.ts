import { App } from 'vue';
import { generateId } from '../../helpers/generateId';
import { defineStore } from 'pinia';

export const useErrorHandlerStore = defineStore('errorHandler', {
    state: () => ({
        recentErrors: {}
    }),
    actions: {
        removeRecentError({ key }: { key: string }): void {
            delete this.recentErrors[key];
        },
        handleError({ err, info }: { err: unknown, info: string }): void {
            console.error(`Got error from '${info}': \n`, err);

            if (Object.keys(this.recentErrors).length >= 2) return;

            const id = generateId();
            this.recentErrors[id] = err;
            window.setTimeout(() => {
                delete this.recentErrors[id];
            }, 25000);
        }
    }
});

export function setUpErrorHandler(app: App<unknown>): void {
    const store = useErrorHandlerStore();
    app.config.errorHandler = (err, vm, info) => {
        store.handleError({ err, info });
    };
}
