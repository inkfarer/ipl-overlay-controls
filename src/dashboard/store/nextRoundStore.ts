import { NextRound } from 'schemas';
import { SetNextRoundRequest } from 'types/messages/rounds';
import { BeginNextMatchRequest } from 'types/messages/activeRound';
import { defineStore } from 'pinia';

const nextRound = nodecg.Replicant<NextRound>('nextRound');

export const nextRoundReps = [ nextRound ];

export interface NextRoundStore {
    nextRound: NextRound;
}

export const useNextRoundStore = defineStore('nextRound', {
    state: () => ({
        nextRound: null
    } as NextRoundStore),
    actions: {
        setShowOnStream(newValue: boolean) {
            nextRound.value.showOnStream = newValue;
        },
        beginNextMatch(request: BeginNextMatchRequest): void {
            nodecg.sendMessage('beginNextMatch', request);
        },
        setNextRound(request: SetNextRoundRequest): void {
            nodecg.sendMessage('setNextRound', request);
        }
    }
});
