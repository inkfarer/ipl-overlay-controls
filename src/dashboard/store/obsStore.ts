import { NodeCGBrowser } from 'nodecg/browser';
import { ObsCredentials, ObsData } from 'schemas';
import { createStore, Store, useStore } from 'vuex';
import cloneDeep from 'lodash/cloneDeep';
import { InjectionKey } from 'vue';
import { SetObsDataRequest } from 'types/messages/obs';

const obsData = nodecg.Replicant<ObsData>('obsData');
const obsCredentials = nodecg.Replicant<ObsCredentials>('obsCredentials');

export const obsReps = [obsData, obsCredentials];

export interface ObsStore {
    obsData: ObsData
    obsCredentials: ObsCredentials
}

export const obsStore = createStore<ObsStore>({
    state: {
        obsData: null,
        obsCredentials: null
    },
    mutations: {
        setState(store, { name, val }: { name: string, val: unknown }): void {
            this.state[name] = cloneDeep(val);
        }
    },
    actions: {
        async connect(store, { address, password }: { address: string, password?: string }): Promise<void> {
            return nodecg.sendMessage('connectToObs', { address, password });
        },
        async setData(store, data: SetObsDataRequest): Promise<void> {
            return nodecg.sendMessage('setObsData', data);
        },
        async startGame(): Promise<void> {
            return nodecg.sendMessage('startGame');
        },
        async endGame(): Promise<void> {
            return nodecg.sendMessage('endGame');
        },
        setEnabled(store, enabled: boolean): void {
            nodecg.sendMessage('setObsSocketEnabled', enabled);
        }
    }
});

export const obsStoreKey: InjectionKey<Store<ObsStore>> = Symbol();

export function useObsStore(): Store<ObsStore> {
    return useStore(obsStoreKey);
}
