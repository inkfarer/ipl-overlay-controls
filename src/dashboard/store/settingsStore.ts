import { LastFmSettings, RadiaSettings, RuntimeConfig } from 'schemas';
import { GameVersion } from 'types/enums/gameVersion';
import { SetGameVersionResponse } from 'types/messages/runtimeConfig';
import { defineStore } from 'pinia';
import { Locale } from 'types/enums/Locale';
import { perGameData } from '../../helpers/gameData/gameData';
import { InterfaceLocale } from 'types/enums/InterfaceLocale';

const lastFmSettings = nodecg.Replicant<LastFmSettings>('lastFmSettings');
const radiaSettings = nodecg.Replicant<RadiaSettings>('radiaSettings');
const runtimeConfig = nodecg.Replicant<RuntimeConfig>('runtimeConfig');

export const settingsReps = [lastFmSettings, radiaSettings, runtimeConfig];

export interface SettingsStore {
    lastFmSettings: LastFmSettings
    radiaSettings: RadiaSettings
    runtimeConfig: RuntimeConfig
}

export const useSettingsStore = defineStore('settings', {
    state: () => ({
        lastFmSettings: {},
        radiaSettings: {
            guildID: null,
            enabled: null,
            updateOnImport: null
        },
        runtimeConfig: null
    } as SettingsStore),
    getters: {
        translatedModeName: state =>
            (mode: string) => (perGameData[state.runtimeConfig.gameVersion]
                .modes[state.runtimeConfig.locale] as Record<string, string>)[mode],
        translatedStageName: state =>
            (stage: string) => (perGameData[state.runtimeConfig.gameVersion]
                .stages[state.runtimeConfig.locale] as Record<string, string>)[stage],
    },
    actions: {
        setLastFmSettings({ newValue }: { newValue: LastFmSettings }): void {
            lastFmSettings.value = newValue;
        },
        setRadiaSettings({ newValue }: { newValue: RadiaSettings }): void {
            radiaSettings.value = newValue;
        },
        setUpdateOnImport(updateOnImport: boolean): void {
            radiaSettings.value.updateOnImport = updateOnImport;
        },
        async attemptRadiaConnection(): Promise<void> {
            return nodecg.sendMessage('retryRadiaAvailabilityCheck');
        },
        setGameVersion(newValue: GameVersion): Promise<SetGameVersionResponse> {
            return nodecg.sendMessage('setGameVersion', { version: newValue });
        },
        setLocale(newValue: Locale): Promise<void> {
            return nodecg.sendMessage('setLocale', newValue);
        },
        setInterfaceLocale(newValue: `${InterfaceLocale}`) {
            runtimeConfig.value.interfaceLocale = newValue;
        }
    }
});
