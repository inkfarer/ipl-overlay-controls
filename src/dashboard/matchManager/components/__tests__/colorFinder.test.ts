import { createTestingPinia, TestingPinia } from '@pinia/testing';
import { config, mount } from '@vue/test-utils';
import { useSettingsStore } from '../../../store/settingsStore';
import { GameVersion } from 'types/enums/gameVersion';
import { useActiveRoundStore } from '../../../store/activeRoundStore';
import ColorFinder from '../colorFinder.vue';

describe('ColorFinder', () => {
    let pinia: TestingPinia;

    config.global.stubs = {
        IplButton: true,
        TeamColorDisplay: true
    };

    beforeEach(() => {
        pinia = createTestingPinia();
        config.global.plugins = [pinia];

        const settingsStore = useSettingsStore();
        // @ts-ignore
        settingsStore.runtimeConfig = { gameVersion: GameVersion.SPLATOON_3 };
        const activeRoundStore = useActiveRoundStore();
        activeRoundStore.swapColorsInternally = false;
    });

    it('matches snapshot', () => {
        const wrapper = mount(ColorFinder, {
            props: {
                colorGroupKey: 'rankedModes'
            }
        });

        expect(wrapper.html()).toMatchSnapshot();
    });

    it('matches snapshot after selecting the first color', async () => {
        const wrapper = mount(ColorFinder, {
            props: {
                colorGroupKey: 'rankedModes'
            }
        });

        await wrapper.get('[data-test="first-color-option_0"]').trigger('click');

        expect(wrapper.html()).toMatchSnapshot();
    });

    it('handles selecting a color', async () => {
        const activeRoundStore = useActiveRoundStore();
        activeRoundStore.setActiveColor = jest.fn();
        const wrapper = mount(ColorFinder, {
            props: {
                colorGroupKey: 'rankedModes'
            }
        });

        await wrapper.get('[data-test="first-color-option_0"]').trigger('click');
        await wrapper.get('[data-test="second-color-option_1"]').trigger('click');
        await wrapper.get('[data-test="apply-button"]').trigger('click');

        expect(activeRoundStore.setActiveColor).toHaveBeenCalledWith({
            categoryKey: 'rankedModes',
            categoryName: 'Ranked Modes',
            color: {
                clrA: '#FC7735',
                clrB: '#4048DB',
                clrNeutral: '#F8F755',
                index: 3,
                isCustom: false,
                key: 'orangeDarkBlue',
                title: 'Orange vs Dark Blue'
            }
        });
        expect(wrapper.emitted('close')).toEqual([[]]);
    });

    it('handles selecting a color when colors are swapped', async () => {
        const activeRoundStore = useActiveRoundStore();
        activeRoundStore.swapColorsInternally = true;
        activeRoundStore.setActiveColor = jest.fn();
        const wrapper = mount(ColorFinder, {
            props: {
                colorGroupKey: 'rankedModes'
            }
        });

        await wrapper.get('[data-test="first-color-option_0"]').trigger('click');
        await wrapper.get('[data-test="second-color-option_1"]').trigger('click');
        await wrapper.get('[data-test="apply-button"]').trigger('click');

        expect(activeRoundStore.setActiveColor).toHaveBeenCalledWith({
            categoryKey: 'rankedModes',
            categoryName: 'Ranked Modes',
            color: {
                clrA: '#4048DB',
                clrB: '#FC7735',
                clrNeutral: '#F8F755',
                index: 3,
                isCustom: false,
                key: 'orangeDarkBlue',
                title: 'Orange vs Dark Blue'
            }
        });
        expect(wrapper.emitted('close')).toEqual([[]]);
    });

    it('does not require selecting a second color if the first option only has one color combination', async () => {
        const activeRoundStore = useActiveRoundStore();
        activeRoundStore.setActiveColor = jest.fn();
        const wrapper = mount(ColorFinder, {
            props: {
                colorGroupKey: 'rankedModes'
            }
        });

        await wrapper.get('[data-test="first-color-option_3"]').trigger('click');
        await wrapper.get('[data-test="apply-button"]').trigger('click');

        expect(activeRoundStore.setActiveColor).toHaveBeenCalledWith({
            categoryKey: 'rankedModes',
            categoryName: 'Ranked Modes',
            color: {
                clrA: '#2C2CDB',
                clrB: '#F29C33',
                clrNeutral: '#FF67EE',
                index: 5,
                isCustom: false,
                key: 'darkBlueOrange',
                title: 'Dark Blue vs Orange'
            }
        });
        expect(wrapper.emitted('close')).toEqual([[]]);
    });

    it('handles cancelling color selection', async () => {
        const wrapper = mount(ColorFinder, {
            props: {
                colorGroupKey: 'rankedModes'
            }
        });

        await wrapper.get('[data-test="cancel-button"]').trigger('click');

        expect(wrapper.emitted('close')).toEqual([[]]);
    });
});
