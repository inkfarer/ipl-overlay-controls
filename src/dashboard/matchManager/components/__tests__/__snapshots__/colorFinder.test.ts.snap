// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ColorFinder matches snapshot 1`] = `
"<div>
  <div class="options-title">translation:colorFinder.firstColorLabel</div>
  <div class="color-finder-options"><button data-v-868771ad="" class="ipl-space color-secondary color-option" data-test="first-color-option_0">
      <div class="color-display" style="background-color: rgb(245, 101, 34);"></div>
    </button><button data-v-868771ad="" class="ipl-space color-secondary color-option" data-test="first-color-option_1">
      <div class="color-display" style="background-color: rgb(237, 217, 33);"></div>
    </button><button data-v-868771ad="" class="ipl-space color-secondary color-option" data-test="first-color-option_2">
      <div class="color-display" style="background-color: rgb(42, 219, 198);"></div>
    </button><button data-v-868771ad="" class="ipl-space color-secondary color-option" data-test="first-color-option_3">
      <div class="color-display" style="background-color: rgb(44, 44, 219); border-color: white;"></div>
    </button><button data-v-868771ad="" class="ipl-space color-secondary color-option" data-test="first-color-option_4">
      <div class="color-display" style="background-color: rgb(179, 222, 69);"></div>
    </button><button data-v-868771ad="" class="ipl-space color-secondary color-option" data-test="first-color-option_5">
      <div class="color-display" style="background-color: rgb(247, 83, 161);"></div>
    </button></div>
  <div class="options-title">translation:colorFinder.secondColorLabel</div>
  <div class="color-finder-options">
    <div data-v-868771ad="" class="ipl-space color-dark color-option disabled">
      <div class="color-display" style="border-color: white;"></div>
    </div>
  </div>
  <team-color-display-stub class="m-t-8"></team-color-display-stub>
  <div class="layout horizontal m-t-8">
    <ipl-button-stub label="translation:colorFinder.applyButton" color="blue" disabled="true" async="false" progressmessage="Working..." successmessage="Done!" errormessage="Error!" requiresconfirmation="false" shortconfirmationmessage="false" small="false" disableonsuccess="false" inline="false" data-test="apply-button"></ipl-button-stub>
    <ipl-button-stub icon="times" color="red" disabled="false" async="false" progressmessage="Working..." successmessage="Done!" errormessage="Error!" requiresconfirmation="false" shortconfirmationmessage="false" small="false" disableonsuccess="false" inline="false" class="m-l-8" title="translation:colorFinder.cancelButtonTitle" data-test="cancel-button"></ipl-button-stub>
  </div>
</div>"
`;

exports[`ColorFinder matches snapshot after selecting the first color 1`] = `
"<div>
  <div class="options-title">translation:colorFinder.firstColorLabel</div>
  <div class="color-finder-options"><button data-v-868771ad="" class="ipl-space color-blue color-option" data-test="first-color-option_0">
      <div class="color-display" style="background-color: rgb(245, 101, 34);"></div>
    </button><button data-v-868771ad="" class="ipl-space color-secondary color-option" data-test="first-color-option_1">
      <div class="color-display" style="background-color: rgb(237, 217, 33);"></div>
    </button><button data-v-868771ad="" class="ipl-space color-secondary color-option" data-test="first-color-option_2">
      <div class="color-display" style="background-color: rgb(42, 219, 198);"></div>
    </button><button data-v-868771ad="" class="ipl-space color-secondary color-option" data-test="first-color-option_3">
      <div class="color-display" style="background-color: rgb(44, 44, 219); border-color: white;"></div>
    </button><button data-v-868771ad="" class="ipl-space color-secondary color-option" data-test="first-color-option_4">
      <div class="color-display" style="background-color: rgb(179, 222, 69);"></div>
    </button><button data-v-868771ad="" class="ipl-space color-secondary color-option" data-test="first-color-option_5">
      <div class="color-display" style="background-color: rgb(247, 83, 161);"></div>
    </button></div>
  <div class="options-title">translation:colorFinder.secondColorLabel</div>
  <div class="color-finder-options"><button data-v-868771ad="" class="ipl-space color-secondary color-option" data-test="second-color-option_0">
      <div class="color-display" style="background-color: rgb(130, 28, 214); border-color: white;"></div>
    </button><button data-v-868771ad="" class="ipl-space color-secondary color-option" data-test="second-color-option_1">
      <div class="color-display" style="background-color: rgb(64, 72, 219); border-color: white;"></div>
    </button></div>
  <team-color-display-stub class="m-t-8" teamacolor="#F56522"></team-color-display-stub>
  <div class="layout horizontal m-t-8">
    <ipl-button-stub label="translation:colorFinder.applyButton" color="blue" disabled="true" async="false" progressmessage="Working..." successmessage="Done!" errormessage="Error!" requiresconfirmation="false" shortconfirmationmessage="false" small="false" disableonsuccess="false" inline="false" data-test="apply-button"></ipl-button-stub>
    <ipl-button-stub icon="times" color="red" disabled="false" async="false" progressmessage="Working..." successmessage="Done!" errormessage="Error!" requiresconfirmation="false" shortconfirmationmessage="false" small="false" disableonsuccess="false" inline="false" class="m-l-8" title="translation:colorFinder.cancelButtonTitle" data-test="cancel-button"></ipl-button-stub>
  </div>
</div>"
`;
